import/core/fable-core.js

```diff
@@ -1987,4 +1987,240 @@
       }, w),
     ];
   };
+  
+  var Event = exports.Event = function(sbscrb, delegates) {
+    var _this = this;
+    this.delegates = delegates || new Array()
+    
+    this.publish = function () {
+       return _this 
+    };
+    
+    this.trigger = function (value) { 
+      Seq.iter(function(f) { f(value) }, _this.delegates ) 
+    };
+    
+    var _addHandler = function (f) {
+      _this.delegates.push(f)
+    };
+    
+    var _removeHandler = function (f) {
+      var fnd = function(el, i, arr) {
+        return ''+el == ''+f //Special dedication to Chet Husk.
+      }
+      
+      var index = _this.delegates.findIndex(fnd)
+      if (index > -1) {
+        _this.delegates.splice(index, 1);
+      }
+    }
+    
+    this.subscribe = function(f) {
+      _addHandler(f);
+    }
+    
+    this.add = function (f) {
+      _addHandler(f);
+    }
+    
+    this.addHandler = function(f) {
+      var h = function(x) {return f(undefined,x) }
+      
+      _addHandler(h);
+    }
+    
+    this.removeHandler =  function(f) {
+      var h = function(x) {return f(undefined,x) }
+     _removeHandler(h); 
+    }
+    
+    this._subscribe = sbscrb || function (observer) {
+      var f = observer.onNext; 
+      _addHandler(f); 
+      
+       var disp = {
+        dispose: function () {
+          _removeHandler(f);
+        }
+      };
+      disp[FSymbol.interfaces] = ["System.IDisposable"];
+      return disp;
+    }
+
+  };
```

Comment:
We could add here `Util.setInterfaces(Event.prototype, ["Microsoft.FSharp.Control.IEvent"]);` so it can be type tested.